source/common/http/header_map_impl.h

```diff
@@ -116,9 +115,7 @@ class HeaderMapImpl : public HeaderMap, NonCopyable {
    * This is the static lookup table that is used to determine whether a header is one of the O(1)
    * headers. This uses a trie for lookup time at most equal to the size of the incoming string.
    */
-  struct StaticLookupTable : public TrieLookupTable<EntryCb> {
-    StaticLookupTable();
-  };
+  struct StaticLookupTable;
```

Comment:
Can you add a doc pointer indicating this is defined in the `header_map_impl.cc` file as opposed to somewhere else?  It might also be useful to replicate the doc comment above to where the class is defined for clarity, and/or more explicitly specify in the doc comment here that TrieLookupTable contains most of the interface for this.

Reply:
OK...done...I feel like the obvious place to look for more detail behind declarations in header_map_impl.h is header_map_impl.cc.

Also this isn't really part of the HeaderMapImpl interface; it's not referenced outside the class, so I didn't want to add a ton more doc to the header file.